src/foam/java/Enum.js

```diff
@@ -1,73 +1,55 @@
-// Test with: foam.u2.ControllerMode.model_.javaSource();
+// Defines template for the EnumModel
```

Comment:
Hi Adam,

I've made foam.java.Enum and extended it from foam.java.Class. I've also updated the general code generation logic to move away from templates and more along the lines of the FOAM2 structure.